modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,375 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background with that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @param array $metadata The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ *
+ * @return array $metadata
+ * @since n.e.x.t
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+
+		return $metadata;
+	}
+
+	$dominant_color = dominant_color_get( $attachment_id );
+
+	if ( ! empty( $dominant_color ) ) {
+		$metadata['dominant_color'] = $dominant_color;
+	}
+
+	return $metadata;
+}
+
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_metadata', 10, 2 );
+
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @param array $metadata Metadata for the attachment.
+ * @param int   $attachment_id attachement id.
+ *
+ * @return array $metadata
+ * @since n.e.x.t
+ */
+function dominant_color_has_transparency_metadata( $metadata, $attachment_id ) {
+
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+		return $metadata;
+	}
+
+	$has_transparency = dominant_color_get_has_transparency( $attachment_id );
+
+	if ( ! empty( $has_transparency ) ) {
+		$metadata['has_transparency'] = $has_transparency;
+	} else {
+		$metadata['has_transparency'] = false;
+	}
```

Comment:
See my other comment. I think it would be better for this function to support returning a `WP_Error`, in which case here we need to only set the value if `! is_wp_error( $dominant_color )`.

Reply:
I want to always set the transparency to be safe the worst result possible is to add a background to a transparent image

Reply:
@pbearne In order to avoid adding a background to a transparent image, we should include such a check when actually putting the background in place, i.e. when modifying the image tags in the frontend, _not_ here when computing the data.

Again, we shouldn't set any metadata that is just a fallback value. We should only set metadata if it has been computed correctly.